Add Navigation events

winsmith opened the pull request:
Add convenience methods to SwiftSDK that allow the easier sending of navigational signals.

Fix #156

Jeehut reviewed (commented):
I just left a comment on the issue with my thoughts on naming. I like the idea though!

See here: https://github.com/TelemetryDeck/web/issues/1491#issuecomment-2168322607

winsmith commented:
> Note that you're still talking about "payload" in the documentation comments, we moved to "parameters" though IIRC.

Noted, thanks a lot! I updated to parameters wherever applicable!
